src/group-constructors.js

```diff
@@ -19,7 +19,50 @@
   }
 
   function constructor(children, timingInput) {
+    if (this._parent === undefined)
+      this._parent = null;
     this.children = children || [];
+    for (c in this.children) {
+      this.children[c]._parent = this;
+    }
+    this._isAncestor = function(effect) {
+      var a = this._parent;
+      while (a !== null) {
+        if (a == effect)
+          return true;
+        a = a._parent;
+      }
+      return false;
+    };
+    this._putChild = function(args, isAppend) {
+      var message = isAppend ? 'Cannot append an ancestor' : 'Cannot prepend an ancestor';
+      for (var i in args) {
+        if (this._isAncestor(args[i])) {
+          throw {
+            type: DOMException.HIERARCHY_REQUEST_ERR,
+            name: 'HierarchyRequestError',
+            message: message
+          };
+        }
+        isAppend ? this.children.push(args[i]) : this.children.unshift(args[i]);
+        args[i]._parent = this;
+      }
+      if (this.timing.duration === 'auto') {
+        this._timing.duration = this.activeDuration;
+      }
+      if (this.animation) {
+        var oldCurrentTime = this.animation._animation.currentTime;
+        this.animation.cancel();
```

Comment:
> oh, the play (github commenting note - comment on the line rather than above for clearer context).
> 
> I assume that play is called to construct the animation or something? Probably need to conditionally call pause afterwards if the animation was paused before appending.

Yep. Will need to add this.